src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -60,6 +64,32 @@ namespace {
 BootReasonType DetermineBootReason()
 {
 #ifdef CONFIG_HWINFO
+
+#ifdef CONFIG_SOC_SERIES_NRF54HX
```

Comment:
Looks like this code doesn't depend on HWINFO, so why is it inside `#ifdef CONFIG_HWINFO`?